registry/remote/example_test.go

```diff
@@ -310,6 +330,35 @@ func ExampleRepository_Fetch_manifestByDigest() {
 	// Example manifest content
 }
 
+func ExampleRepository_Fetch_referenceManifest() {
+	reg, err := remote.NewRegistry(host)
+	if err != nil {
+		panic(err)
+	}
+	ctx := context.Background()
+	repo, err := reg.Repository(ctx, exampleRepositoryName)
+	if err != nil {
+		panic(err)
+	}
```

Comment:
I am thinking since we don't need to access `reg`, should we use `remote.NewRepository` directly? 

This is beyond the scope of adding a reference manifest example but will prettify repository-related examples.

Reply:
Changed all relevant occurences to `remote.NewRepository`